lib/blocs/tasks_bloc.dart

```diff
@@ -54,29 +57,57 @@ class _TasksBloc {
           .replaceAll('{np}', ' ### $negativePrompt');
       print(formattedPrompt);
 
-      final json = {
-        'prompt': formattedPrompt,
-        'params': {
-          'steps': 30,
-          'n': 1,
-          'sampler_name': 'k_euler',
-          'width': 512,
-          'height': 512,
-          'cfg_scale': 7,
-          'seed_variation': 1000,
-          'seed': seed == null ? '' : '$seed',
-          'karras': true,
-          //'denoising_strength': mutationRate,
-          'post_processing': postProcessors,
-        },
-        'nsfw': false,
-        'censor_nsfw': false,
-        'trusted_workers': false,
-        //'source_processing': 'img2img',
-        //'source_image': base64.encode(sourceImage.buffer.asUint8List()),
-        'models': [modelName],
-        'r2': true,
-      };
+      final Map<String, dynamic> json;
+
+      if (isLoggedIn && img2ImgInputEncodedString != null) {
```

Comment:
It seems like programmer error if we end up here with img2ImgInputEncodedString != null, and isLoggedIn == false, right?

If so, I would fail fast with:

```dart
if (img2ImgInputEncodedString != null && !isLoggedIn) {
   throw Exception('blah blah');
}
```

Otherwise, this code could incorrectly request txt2img, when the user is expecting img2img

Reply:
Ah got it, yeah that makes sense.